internal/pkg/action/operator_list_custom_resources.go

```diff
@@ -0,0 +1,35 @@
+package action
+
+import (
+	"context"
+
+	lib "github.com/operator-framework/kubectl-operator/pkg/action"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// OperatorListCustomResources knows how to find and list custom resources given a package name and namespace.
+type OperatorListCustomResources struct {
+	config      *Configuration
+	PackageName string
+}
+
+func NewOperatorListCustomResources(cfg *Configuration) *OperatorListCustomResources {
+	return &OperatorListCustomResources{
+		config: cfg,
+	}
+}
+
+func (o *OperatorListCustomResources) Run(ctx context.Context) (*unstructured.UnstructuredList, error) {
+	opKey := types.NamespacedName{
+		Name:      o.PackageName,
+		Namespace: o.config.Namespace,
+	}
+
+	result, err := lib.ListAll(ctx, o.config.Client, opKey)
```

Comment:
The intent of `pkg/action` is to provide the main Go-library style entrypoints for the high level functions provided by kubectl-operator's CLI, such that other clients (e.g. console) can use them whole-cloth without having to think about the internals.

That generally means the only exported API in this package should be the structs used by the CLI and their `Run` functions. Anything else should be the exception rather than the rule.

At a minimum, helper functions like this should be put in separate packages more aligned with their concern. If they're only relevant for one particular action (like this one is), I think we should leave it right here in this file and unexport it.

If and when we see that there is code duplication or opportunity for re-use, then that's what I would say we should move functions to a helper package.